controllers/topology/cluster_controller.go

```diff
@@ -133,12 +133,43 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return ctrl.Result{}, nil
 	}
 
+	patchHelper, err := patch.NewHelper(cluster, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	// Create a scope initialized with only the cluster; during reconcile
 	// additional information will be added about the Cluster blueprint, current state and desired state.
-	scope := scope.New(cluster)
+	s := scope.New(cluster)
+
+	defer func() {
+		// Use the desired cluster if it is available.
+		// If computing desired state failed we wont have a desired cluster, in that case use the current cluster.
+		var cluster *clusterv1.Cluster
+		if s.Desired != nil && s.Desired.Cluster != nil {
+			cluster = s.Desired.Cluster
+		} else {
+			cluster = s.Current.Cluster
+		}
```

Comment:
can we just take cluster (that is the current cluster)?

Reply:
I think current cluster all the time should be okay. I will test and update accordingly.

Reply:
using `cluster` always works 👍🏼